controller.go

```diff
@@ -43,24 +43,30 @@ func SyncPVC(pvc *PVClaim) {
 				// No PV could be found
 				// OBSERVATION: pvc is "Pending", will retry
 				if hasAnnotation(pvc, annClass) {
-					plugin := findProvisionerPluginForPV(pv) // Need to flesh this out
-					if plugin != nil {
-						//FIXME: left off here
-						// No match was found and provisioning was requested.
-						//
-						// maintain a map with the current provisioner goroutines that are running
-						// if the key is already present in the map, return
-						//
-						// launch the goroutine that:
-						// 1. calls plugin.Provision to make the storage asset
-						// 2. gets back a PV object (partially filled)
-						// 3. create the PV API object, with claimRef -> pvc
-						// 4. deletes itself from the map when it's done
-						// return
-					} else {
-						// make an event calling out that no provisioner was configured
-						// return, try later?
+					// User requested dynamic provisioning.
+					// provisionClaim() starts a goroutine to provision a PV
+					// for this claim.
+					// Possible outcomes:
+					// - if it succeeds, next syncClaim() will find the
+					//   provisioned PV and will bind the claim to it.
+					// - if it fails, next syncClaim() will call
+					//   provisionClaim() again, which will retry provisioning.
+					// - if it takes too long (or someone modifies the PVC),
+					//   syncClaim() will call provisionClaim() again.
+					//   provisionClaim() must be smart enough not to start new
+					//   provisioning for PVC if there is one already in
+					//   progress.
+					// - if the controller crashes and is restarted, we loose
```

Comment:
I am a little concerned about this if the controller goes down, or there's a panic in another controller, but the only way to do anything about that is to serialize the state somewhere, which is basically opening a new can of worms.  I don't think there's any way, in a vacuum, to 100% prevent PV leaks due to concerns like this, so I suggest that we all it out as a fact of life for now.

@jsafrane @saad-ali